### ISSUE TITLE
BUILD ERROR if unrelated package.json exists in a sub-folder

### ISSUE THREAD

**Issue by cogitaria-admin** [dherges/ng-packagr#213]
## Type of Issue

```
[x ] Bug Report
[ ] Feature Request
```

## Description

Added a folder called **server** at the root and containing a ten-line Express application and of course an accompanying package.json.  The ng-packagr build then immediately fails.  

BUILD ERROR
Error File '/Users/admin/src/lytx/ngx/.ng_pkg_build/ngx-server/ts/server/public_api.ts' not found.
Error: Error File '/Users/admin

In fact if you delete **everything** in ./server except package.json you get this failure.  

#### How To Reproduce

See description

#### Expected Behaviour

Ignore unrelated sub-folders. 

#### Version Information

```
ng-packagr:  "^1.5.0-rc.0",

@angular: 4.2.4

```

**Comment by DavidParks8** [comment]
The secondary entry mechanism is designed such that all package.json files below the root ng-package.json are treated as secondary entries. It is still possible to have your server in the same repo by locating your ng-package.json in a folder structure such as:
```
/root
    /server
        express app
    /lib
        **/*.ts
        ng-package.json
        package.json -- this would be the package.json that gets distributed with your library
    package.json -- this is the package.json with your scripts
```

And then your scripts would include
```json
scripts: {
  "build:lib": "ng-packagr -p lib/ng-package.json"
}
```

**Comment by DavidParks8** [comment]
I wonder if your issue could also be resolved by reading package.json files below the entry ts file instead of the root ng-package.json file. Would that work with your folder structure?

**Comment by dherges** [comment]
In relation to #190, this would be an argument for not relying on `package.json` or at least validating if `ngPackage` property is present in the package.json? :question:

**Comment by dherges** [comment]
I understand the idea behind the auto-discovery of the secondary entries. However, technically, the new feature is now breaking existing builds when the `ng-package.json` was located in the root folder and we now pick up unrelated `package.json` files.

**Comment by dherges** [comment]
If I see right, the secondaries are discovered under the `src` directory.

https://github.com/dherges/ng-packagr/blob/9de75241aba6bcf511d88a465e8ffbc5bf59b03c/src/lib/steps/package.ts#L77-L88

This is why vmware/clarity#1608 works because clartiy sets the `src` property in their `ng-package.json`

**Comment by dherges** [comment]
For comparison, the first approach (never finalized / implemented) with explicit configuration for secondaries:

**Comment by DavidParks8** [comment]
I propose we make this feature more configurable. We could offer 3 options for secondaries.
```json
"secondaryEntrypoints": "none" // this would be the default
...
"secondaryEntrypoints: "automatic" // this would mimic the behavior in 1.5.0-rc.1
...
"secondaryEntrypoints: ": [  // this would behave as you've described
   { 
     "name": "bar",  // I'm not sold on this field because the name must match the folder paths in dist, and it may cause complications and undesired behaviors. 
     "entryFile": "bar/public_api.ts" // If we require this to be a subfolder (like the automatic behavior) then we could easily generate package names
   }, 
   { 
     "name": "baz", 
     "entryFile": "baz/public_api.ts" 
   }
]
```

With such configurability, people can choose to implement their libraries in the way they want.

**Comment by DavidParks8** [comment]
If we don't want the overhead of maintaining multiple configuration capabilities, then validating the presence of an `ngPackage` field in secondary package.json files would be an easy and reasonable fix to apply.

**Comment by DavidParks8** [comment]
@dherges whatever you decide, I would be happy to make the changes in a PR.

**Comment by cogitaria-admin** [comment]
Thanks for explanations and workarounds.

**Comment by dherges** [comment]
As a **fix** for the 1.5.0 feature release:

- for auto-discovery of secondary entries, validate whether the `package.json` has a `ngPackage` property (can be an empty object)

For later releases, tracked by #190
